lib/v2/twitter/web-api/twitter-api.js

```diff
@@ -232,7 +233,45 @@ const cacheTryGet = async (cache, _id, params, func) => {
     return cache.tryGet(`twitter:${id}:${funcName}:${paramsString}`, () => func(id, params), config.cache.routeExpire, false);
 };
 
-const getUserTweets = (cache, id, params = {}) => cacheTryGet(cache, id, params, getUserTweetsByID);
+// returns:
+// 1. nothing for some users
+// 2. HOT tweets for the other users, instead of the LATEST ones
+const _getUserTweets = (cache, id, params = {}) => cacheTryGet(cache, id, params, getUserTweetsByID);
+// workaround for the above issue:
+// 1. getUserTweetsAndReplies return LATEST tweets and replies, which requires filtering
+//    a. if one replies a lot (e.g. elonmusk), there is sometimes no tweets left after filtering, caching may help
+// 2. getUserMedia return LATEST media tweets, which is a good plus
+const getUserTweets = async (cache, id, params = {}) => {
+    let tweets = [];
+    const rest_id = await getUserID(cache, id);
+    await Promise.all(
+        [_getUserTweets, getUserTweetsAndReplies, getUserMedia].map(async (func) => {
+            try {
+                tweets.push(...(await func(cache, id, params)));
+            } catch (e) {
+                logger.warn(`Failed to get tweets for ${id} with ${func.name}: ${e}`);
+            }
+        })
+    );
+
+    const cacheKey = `twitter:user:tweets-cache:${rest_id}`;
+    let cacheValue = await cache.get(cacheKey);
+    if (cacheValue) {
+        cacheValue = JSON.parse(cacheValue);
+        if (cacheValue && cacheValue.length) {
+            tweets = cacheValue.concat(tweets);
+        }
+    }
+    const idSet = new Set();
+    tweets = tweets
+        .filter((tweet) => !tweet.in_reply_to_screen_name) // exclude replies
+        .map((tweet) => (idSet.has(tweet.id_str) ? null : idSet.add(tweet.id_str) && tweet)) // deduplicate
```

Comment:
https://deepscan.io/dashboard/#view=project&tid=6244&pid=8135&bid=92448&subview=pull-request&prid=1676556
```suggestion
        .map((tweet) => (idSet.has(tweet.id_str) ? null : idSet.add(tweet.id_str)) && tweet) // deduplicate
```